worker/src/main/java/gov/cms/ab2d/worker/adapter/bluebutton/BeneficiaryAdapterStub.java

```diff
@@ -0,0 +1,112 @@
+package gov.cms.ab2d.worker.adapter.bluebutton;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.util.Assert;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.time.Month;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * This is a stub implementation that we can use till the BFD API becomes available.
+ * The rightmost 3 characters of the contractNumber being passed in must be numeric.
+ */
+@Slf4j
+@Component
+public class BeneficiaryAdapterStub implements BeneficiaryAdapter {
+
+    private static final String BENE_ID_FILE = "/test-stub-data/synthetic-bene-ids.csv";
+
+
+    @Override
+    public GetPatientsByContractResponse getPatientsByContract(String contractNumber) {
+
+        final int contractSno = extractContractSno(contractNumber);
+        final int startOffset = contractSno * 100;
+
+        final var patientsPerContract = getFromSampleFile(startOffset);
+
+        return toResponse(contractNumber, patientsPerContract);
+    }
+
+
+    private Integer extractContractSno(String contractNumber) {
+        final String contractNumberSuffix = contractNumber.substring(contractNumber.length() - 3);
+
+        final Integer sno;
+        try {
+            sno = Integer.valueOf(contractNumberSuffix);
+        } catch (NumberFormatException e) {
+            final String errMsg1 = String.format("Invalid ContractNumber : %s.  ", contractNumber);
+            final String errMsg2 = "The rightmost 3 characters of the contract number must be numeric. ";
+            throw new IllegalArgumentException(errMsg1 + errMsg2);
```

Comment:
I'd say, for any contract number that does not fall into S0001--S0300 pattern, just return an empty list of benes (since this is what BFD would do). No need to throw IAE.

Reply:
done